src/main/java/org/folio/inventory/instanceingress/handler/CreateInstanceIngressEventHandler.java

```diff
@@ -0,0 +1,240 @@
+            )
+            .withParsedRecord(new ParsedRecord()
+              .withId(event.getId())
+              .withContent(event.getEventPayload().getSourceRecordObject())
+            );
+          event.getEventPayload()
+            .withAdditionalProperty(MARC_BIBLIOGRAPHIC.value(), marcBibRecord);
+
+          LOGGER.info("Manipulating fields of a Record from InstanceIngressEvent with id '{}'", event.getId());
+          var mappingParameters = Json.decodeValue(mappingMetadata.getMappingParams(), MappingParameters.class);
+          AdditionalFieldsUtil.updateLatestTransactionDate(marcBibRecord, mappingParameters);
+          AdditionalFieldsUtil.move001To035(marcBibRecord);
+          AdditionalFieldsUtil.normalize035(marcBibRecord);
+          AdditionalFieldsUtil.addFieldToMarcRecord(marcBibRecord, TAG_035, SUBFIELD_A, event.getEventPayload().getSourceRecordIdentifier() + BIBFRAME);
+
+          LOGGER.info("Mapping a Record from InstanceIngressEvent with id '{}' into an Instance", event.getId());
+          var parsedRecord = new JsonObject((String) marcBibRecord.getParsedRecord().getContent());
+          RecordMapper<org.folio.Instance> recordMapper = RecordMapperBuilder.buildMapper(MARC_BIB_RECORD_FORMAT);
+          var instance = recordMapper.mapRecord(parsedRecord, mappingParameters, new JsonObject(mappingMetadata.getMappingRules()));
+          LOGGER.info("Mapped Instance from InstanceIngressEvent with id '{}': {}", event.getId(), instance);
+          return super.idStorageService.store(event.getId(), instance.getId(),
+              context.getTenantId())
+            .map(r -> instance)
+            .onFailure(e -> LOGGER.error("Error creating relationship of inventory recordId '{} and instanceId '{}'", event.getId(), instance.getId()));
+        } catch (Exception e) {
+          LOGGER.warn("Error during preparing and executing mapping:", e);
+          return Future.failedFuture(e);
+        }
+      });
+  }
+
+  private Future<Instance> validateInstance(org.folio.Instance instance, InstanceIngressEvent event) {
+    try {
+      LOGGER.info("Validating Instance from InstanceIngressEvent with id '{}':", event.getId());
+      var instanceAsJson = prepareInstance(instance, event.getEventPayload().getSourceType());
+      var errors = EventHandlingUtil.validateJsonByRequiredFields(instanceAsJson, INSTANCE_REQUIRED_FIELDS);
+      return failIfErrors(errors, event.getId())
+        .orElseGet(() -> {
+          var mappedInstance = Instance.fromJson(instanceAsJson);
+          var uuidErrors = ValidationUtil.validateUUIDs(mappedInstance);
+          return failIfErrors(uuidErrors, event.getId())
+            .orElseGet(() -> Future.succeededFuture(mappedInstance));
+        });
+    } catch (Exception e) {
+      return Future.failedFuture(e);
+    }
+  }
+
+  private static Optional<Future<Instance>> failIfErrors(List<String> errors, String eventId) {
+    if (!errors.isEmpty()) {
```

Comment:
CollectionUtils.isNotEmpty
but bettor to inverse it
```java
if (isEmpty(errors) {
    return Optional.empty()
}
//some logic without bracket on the top level of line
```

Reply:
fixed, thank you